filebeat/module/elasticsearch/slowlog/test/test.log-expected.json

```diff
@@ -12,7 +12,9 @@
         "elasticsearch.slowlog.took_millis": 4, 
         "elasticsearch.slowlog.total_hits": 19435, 
         "elasticsearch.slowlog.total_shards": 1, 
-        "elasticsearch.slowlog.types": "", 
+        "elasticsearch.slowlog.types": [
+            ""
+        ], 
```

Comment:
I wonder if we want to index an array of one element, that element being an empty string, over here OR if we want o index an empty array OR not even index this field if no types are present.

Now that Ingest processors support `if` conditions (https://github.com/elastic/elasticsearch/pull/32398), perhaps we could define the `split` processor like so:

```json
{
   "split":{
      "field":"elasticsearch.slowlog.types",
      "separator":",",
      "ignore_missing":true,
      "if":"ctx.elasticsearch.slowlog.types != ''"
   }
}
```

Reply:
@ruflin any opinion on that?
IMHO `types` field should be stored since it appears in the log line.
Arrays and single values are internally stored the same way since any ES field can be an array. Indeed _source will look different since source document is different, but should we care?

Reply:
The log entry that this is coming from looks like this: `total_hits[19435], types[], stats[],`. If I run the following command in DevTools what I get is `"a": []`. So I would argue `types` should be there but with an empty array and not entry.

```
POST test/_doc
{
  "a": []
}
```